### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by JanZachmann**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
 [dependencies]
 async-trait = "0.1"
-bluer = { version = "0.15", features = ["bluetoothd"] }
-clap = { version = "3.2", features = ["derive"] }
+bluer = { version = "0.17", features = ["bluetoothd"] }
+clap = { version = "4.5", features = ["derive"] }
 enclose = "1.1"
-env_logger = "0.8"
+# ToDo: check if 'default-features = false' can be removed. currently needed as workaround for cargo-bitbake.
+env_logger = { version = "0.11", default-features = false }
```
Comment (mlilien) [inline Cargo.toml:19]: what does it mean? which functionality changed?

File: wifi-commissioning-gatt-service_0.4.2.bb [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# Auto-Generated by cargo-bitbake 0.3.16
```
Comment (mlilien) [inline wifi-commissioning-gatt-service_0.4.2.bb:1]: rm file

**Review by mlilien — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (JanZachmann) [reply Cargo.toml:19]: we want to update env_logger in order to get rid of `atty` audit warning.

if we use 0.11 with default features I get the following:
```
jan@JZ-Mint-I631:~/projects/wifi-commissioning-gatt-service$ cargo-bitbake bitbake
error: failed to select a version for `env_logger`.
    ... required by package `wifi-commissioning-gatt-service v0.4.2 (/home/jan/projects/wifi-commissioning-gatt-service)`
versions that meet the requirements `^0.11` are: 0.11.3

the package `wifi-commissioning-gatt-service` depends on `env_logger`, with features: `anstream` but `env_logger` does not have these features.
 It has an optional dependency with that name, but but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.


failed to select a version for `env_logger` which could resolve this conflict
```

Comment (mlilien) [reply Cargo.toml:19]: which functionality changes when we use the crate without default features? if nothing, it seems we don't need this default features anyway.

**Review by mlilien — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (JanZachmann) [reply Cargo.toml:19]: https://github.com/rust-cli/env_logger/blob/0e25d9e993241f1c57f6f7c2ab775a407a87fa3f/Cargo.toml#L119C1-L119C47

at least it still compiles. so yes I guess we don't need the features at all

**Pull request merged**